backport-2.0: ui: Auto-zoom map to fit visible content

mrtracy opened the pull request:
The cluster map now auto-zooms to the set of visible information
whenever the user navigates to a new locality, or whenever new locality
information becomes available for the locality currently being viewed.

Fixes #23513

Release note: None

Commit:
ui: Auto-zoom map to fit visible content

The cluster map now auto-zooms to the set of visible information
whenever the user navigates to a new locality, or whenever new locality
information becomes available for the locality currently being viewed.

Fixes #23513

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/23875)
<!-- Reviewable:end -->

vilterp commented:
LGTM, but let's hold off until we unbreak cluster viz by verifying, merging and cherry-picking #23879

Commit:
ui: Animate zoom to contents

Animates the zoom-to-content transition using d3. Refactors some of the
zooming logic to clarify the interaction between d3's zoom functionality
and our react state.

Also addresses some outstanding code review comments from #23866

Resolves #23878

Release note: None

vilterp reviewed (approved):
K, fix is merged, so we're good to go on this.

couchand commented:
OTOH i raised some questions on the PR this cherry-picks that I think might be worth addressing.

vilterp commented:
sure. up to @mrtracy whether to address them here or in his other zoom-related PR that hasn't been 
merged to master yet.

mrtracy commented:
This cherry-pick now includes the follow-up commit #23883, which animates the transition.  This is ready for inclusion into 2.0 or 2.0.1, as deemed appropriate.

bdarnell commented:
Let's merge this for 2.0; it'll be good for demos and is isolated from anything else.
